src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepository.cs

```diff
@@ -746,13 +750,13 @@ public ITransactionContext BeginTransaction(string walletName)
                 else
                     walletContainer = new WalletContainer(this.GetConnection(walletName), null, this.processBlocksInfo);
 
-                walletContainer.LockUpdateWallet.Wait();
```

Comment:
`LockUpdateWallet` should not be used directly. It circumvents some other important logic found in `WriteLockWait`.